schema/schema_test.go

```diff
@@ -0,0 +1,183 @@
+package schema_test
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/hpcloud/termui/schema"
+
+	term "github.com/cloudfoundry/cli/cf/terminal"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+// tests for schema
+func TestSchemaString(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	RunSpecs(t, "Schema")
+}
+
+var _ = Describe("Schema", func() {
+	Describe("Test if schema asks correctly for a string", func() {
+		Context("ask for string", func() {
+			It("should be Insert string value for /stringtest [required]", func() {
+				ui := term.NewUI(os.Stdin, term.NewTeePrinter())
+
+				str := `{
+   "type":"object",
+   "properties":{
+      "stringtest":{
+         "type":"string"
+      }
+   },
+   "required":[
+      "stringtest"
+   ]
+}`
+
+				old := os.Stdout
+				r, w, _ := os.Pipe()
+				os.Stdout = w
+
+				schemaParser := schema.NewSchemaParser(ui)
+				schemaParser.ParseSchema(str)
+
+				outC := make(chan string)
+
+				go func() {
+					var buf bytes.Buffer
```

Comment:
We can get rid of this goroutine via `ioutil.ReadAll`:
```go
old := os.Stdout
defer func() { os.Stdout = old }()
r, w, _ := os.Pipe()
os.Stdout = w

schema.NewSchemaParser(ui).ParseSchema(str)

w.Close()
out, err := ioutil.ReadAll(r)
Expect(err).NotTo(HaveOccurred())
Expect(string(out)).To(ContainSubstring("Insert string value for /stringtest [required]"))
```

(I'm sad that `os.Stdout` needs to be an `*os.File` and we can't just do `os.Stdout = bytes.Buffer{}`)